pyscf/adc/test/test_radc/test_H2O_radc_outcore.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2014-2019 The PySCF Developers. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Author: Samragni Banerjee <samragnibanerjee4@gmail.com>
+#         Alexander Sokolov <alexander.y.sokolov@gmail.com>
+#
+
+import unittest
+import numpy
+import math
+from pyscf import gto
+from pyscf import scf
+from pyscf import adc
+
+mol = gto.Mole()
+r = 0.957492
+x = r * math.sin(104.468205 * math.pi/(2 * 180.0))
+y = r * math.cos(104.468205* math.pi/(2 * 180.0))
+mol.atom = [
+    ['O', ( 0., 0.    , 0)],
+    ['H', ( 0., -x, y)],
+    ['H', ( 0., x , y)],]
+mol.basis = {'H': 'aug-cc-pVDZ',
+             'O': 'aug-cc-pVDZ',}
+mol.verbose = 0
+mol.build()
+
+mf = scf.RHF(mol)
+mf.conv_tol = 1e-12
+mf.kernel()
+myadc = adc.ADC(mf)
+
+def tearDownModule():
+    global mol, mf
+    del mol, mf
+
+class KnownValues(unittest.TestCase):
+
+
+    def test_ea_adc2(self):
+
+        myadc.max_memory = 20
+        e, t_amp1, t_amp2 = myadc.kernel_gs()
+        self.assertAlmostEqual(e, -0.2218560609876961, 6)
+
+        
+        myadcea = adc.radc.RADCEA(myadc) 
+        e,v,p = myadcea.kernel(nroots=3)
+
+        self.assertAlmostEqual(e[0], 0.0287675413010661, 6)
+        self.assertAlmostEqual(e[1], 0.0553475511361251, 6)
+        self.assertAlmostEqual(e[2], 0.1643553780332306, 6)
+
+        self.assertAlmostEqual(p[0], 1.9868196915945326, 6)
+        self.assertAlmostEqual(p[1], 1.9941128865405613, 6)
+        self.assertAlmostEqual(p[2], 1.9760420333383126, 6)
+
+
+    def test_ea_adc3(self):
+
+        myadc.method = "adc(3)"
+        myadc.incore_complete = False
```

Comment:
Is setting incore_complete = False sufficient to guarantee that the code is out of core?

Looking at your code below
`        if (self._scf._eri is not None and
             (mem_incore+mem_now < self.max_memory or self.incore_complete)):
`
it seems that we can have incore_complete = False but if (mem_incore+mem_now < self.max_memory) is True and we can still run incore.

Reply:
No, setting incore_complete = False sufficient to guarantee that the code is out of core. I added max_memory.